crates/ty_server/src/server/api/requests/workspace_diagnostic.rs

```diff
@@ -71,6 +136,53 @@ impl BackgroundRequestHandler for WorkspaceDiagnosticRequestHandler {
 
         Ok(reporter.finish())
     }
+
+    fn process(
+        id: &RequestId,
+        snapshot: SessionSnapshot,
+        client: &Client,
+        params: WorkspaceDiagnosticParams,
+    ) {
+        // if streaming: it's a partial result if we did send any changes to the client but a regular report otherwise.
+        // for non-streaming, it's always a full report.
+        // We can simply test if all items are unchanged
+        let result = Self::run(&snapshot, client, params.clone());
```

Comment:
This comment doesn't seem relevant to what's happening with the code which simply calls `run`, is this intended to be placed somewhere else?